### PR TITLE
Disclose additional registrant data

### LINKED ISSUE #109: Disclose additional registrant data
Registrant requested an option to disclose additional contact data on the public whois

* email address
* phone nr

currently email address is already public for org registrants but only via internet.ee rest-whois (behind captcha).

add additional disclosing options to publish email address and phone number of a registrant on whois without captcha protection

Also requires changes in whois, rest-whois and registrant API @ registry

### PR DESCRIPTION
**Pull request by thiagoyoussef**
close #109

### CODE REVIEW

File: src/translations/et.json [hunk 1]
```diff
@@ -238,11 +238,15 @@
...
   "whois.domain.detailInfo": "Detailne info",
+  "whois.public.text": "Avalikult kättesaadav",
+  "whois.public.tooltip": "Kontaktandmed muutuvad vaikimisi kättesaadavaks ainult internet.ee lehel captcha checki taga, et vältida sinu andmete kogumist. See valik muudab andmed kõikjal kättesaadavaks.",
   "whois.edit.checkAll": "({count}/{countOf}) Kõik info nähtav",
   "whois.edit.checkNone": "({count}/{countOf}) Kõik info peidetud",
   "whois.edit.checkSome": "({count}/{countOf}) Osa info peidetud",
+  "whois.edit.phone": "Telefon avalik ({contactPhone})",
   "whois.edit.email": "E-mail avalik ({contactEmail})",
   "whois.edit.name": "Nimi avalik ({contactName})",
+  "whois.edit.registrant_publishable": "Jah, ma luban oma andmed ilma captcha kontrollita kättesaadavaks teha",
```
Comment (thiagoyoussef) [inline src/translations/et.json:249]: Is the estonian translation correct?

File: src/components/WhoIsEdit/WhoIsEdit.jsx [hunk 2]
```diff
@@ -24,9 +24,8 @@ function WhoIsEdit({ contacts, isOpen, checkAll, onChange, domain }) {
                     }
                     return disclosed_attributes.size;
                 })(),
-            isCheckedAll:
-                domain.registrant.org || disclosed_attributes.size === contactsList.length * 2,
-            totalCount: acc.totalCount + 2,
+            isCheckedAll: disclosed_attributes.size === contactsList.length * 3,
```
Comment (OlegPhenomenon) [inline src/components/WhoIsEdit/WhoIsEdit.jsx:27]: what differents between size and length?
what number does size start counting from? 0 or 1? :)

**Review by OlegPhenomenon — APPROVED** [review]

**Comment by vohmar** [comment]
deploly to staging failed
https://gist.github.com/ratM1n/e44514439993d7cffb40fcd71eb06411

**Review by OlegPhenomenon — APPROVED** [review]

**Comment by vohmar** [comment]
* by default all disclose settings seem to be turned to 
  * name public
  * email public
  * that is not correct - by default all private contact data should be undisclosed and only name of the org type contact should be disclosed
    * i registered a new privately owned domain
    * logged in to registrant as admin contact of that domain and as a result the admin contact name and email became disclosed in public whois
    * looking at the contact disclose data i see that name and email should be public but as without captcha is disabled i presume that this data is available only after solving captcha check
* after disclosing phone nr email was hidden from public view - so logging into registrant portal sets some invalid set visibility to the user
* when i undisclosed the data i get the notification that no domain registrations are affected by the change but whois data was correctly updated
* when registrant disclosed its contact data the same data was disclosed for admin and tech as well even though these were different objects and different persons
  * the disclosing and undisclosing must affect only that particular object
* undisclosed all data for a domain including removing captcha check got again blank affected domains message
  * name of the private registrant became available without captcha requirement - should not be as the allow without captcha was not checked
* allow my data to be available without captcha setting reveals all the data of the contacts no just the disclosed data set by the user for that contact
  * it should make public only the data that is checked for that contact to be visible

**Review by OlegPhenomenon — APPROVED** [review]

**Comment by vohmar** [comment]
Rechecked i hope most of use cases and these are the findings:
* in most cases the warning popup when changing disclose settings is empty. Usually ok on the first action, but later on other changes will be shown empty - no matter if there is only one domain related to that object or multiple
* if contact is also associated to another locked domain the disclose attributes of that domain do not seem to be updated (in ui at least)
* private registrant name is revealed without captcha regardless of the state of "available without captcha" - should not be
* if user has multiple roles with different objects on the domain all disclose options are displayed separately including the "available without captcha" but the captcha option works only for registrant object (t2itsakatki.ee: Mary-Änn = rant, tech). 
  * non-registrant no captcha publish option produces error in "change contacts" view - "error in new data..." pointing to phone nr

**Comment by vohmar** [comment]
org owned registrations:
* by default registrant name, reg no and country code should be disclosed without captcha
* registrant email, last update; admin's name, email, last update; tech name, email, last update are disclosed after captcha check
* @satging
  * regisntrant name is hidden behind captcha
  * last update of admin and tech contacts is not disclosed
  * removed captcha requirement from registrant data (phone disclosed) - in whoid phone still hidden behind captcha. not an issue with other roles including with the same object

**Pull request merged**